### PR TITLE
AE Explorer V3.0.0

### LINKED ISSUE #10: Add ToolTips

### LINKED ISSUE #11: Show # of Enrolled/Randomized Participants
Not sure the best way to handle this, but it would be good to have the number of relevant study participants shown somewhere on the chart. Maybe add a footnote for "X of X participants shown in this view"

### LINKED ISSUE #16: Allow generation of Details table from summary row
See #3 for related request.

### LINKED ISSUE #17: Refactor using modular code with ES6
And then implement a build process using node/npm/rollup/babel similar to the one described in [webcharts-wrappers-boilerplate](https://github.com/RhoInc/webcharts-wrapper-boilerplate)

### LINKED ISSUE #21: Combine Settings and Variables object
Biggest remaining hurdle is the settings.groups object, but that can be generated automatically. The chartfoundry renderer already does something [similar](https://github.com/RhoInc/chf-renderer-aeexplorer/blob/master/src/Renderer.js#L58-L59).

### LINKED ISSUE #26: Diff column displaying even when defaults.diffCol is set to 'Hide'.

### LINKED ISSUE #27: Group columns not sorting alphabetically.

### LINKED ISSUE #28: Add event frequencies and add group column mouseover event.

### PR DESCRIPTION
**Pull request by samussiah**
@brittsikora @jwildfire Refactor complete: http://bl.ocks.org/samussiah/raw/90e524b9ef2fdf8214d491aaca4cd3b3/

Closes #10 
Closes #11 
Closes #16 
Closes #17 
Closes #21 
Closes #26 
Closes #27 
Closes #28

### CODE REVIEW

**Comment by brittsikora** [comment]
#10 - looks good
#11 - n makes sense - pass
#16 - details table works when a summary item is selected
#28 - event frequencies work when a group's event is moused-over

Commented on the following issues for @jwildfire to help me figure out how to test them:
#17 
#21 
#26 
#27

**Comment by jwildfire** [comment]
@brittsikora @samussiah #17 needs a quick code review to confirm. All others are close to done and just need some final housekeeping. My comments are on the issues.

**Comment by jwildfire** [comment]
@brittsikora @samussiah We're also going to need to update the wiki for the library before we accept this pull request and move to v3.0. Is that something that the 2 of your can collaborate on?

**Comment by samussiah** [comment]
Wiki updated: https://github.com/RhoInc/aeexplorer/wiki

File: src/init.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*------------------------------------------------------------------------------------------------\
+  Initialize adverse event explorer.
+\------------------------------------------------------------------------------------------------*/
+
+export function init(data) {
+    var element = d3.select(this.element);
```
Comment (jwildfire) [inline src/init.js]: would it be more clear sense to call this `selector`?

File: src/init.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    var settings = this.config;
+
+  //Render single column if no group variable is specified.
+    if (!(settings.variables.group) || ['', ' '].indexOf(settings.variables.group) > -1) {
+        settings.variables.group = 'data_all';
+        settings.defaults.totalCol = '';
+        settings.groups = [{'key':'All'}];
+    }
+
+    function errorNote(msg) {
```
Comment (jwildfire) [inline src/init.js]: Lets move the settings checks to a separate method in a follow up release. Created #33.

File: src/init.js (hunk 1 above)
Comment (jwildfire) [inline src/init.js]: or perhaps `canvas` (which is used elsewhere). Idea is to differentiate between the css selector (which is a string) and the d3 selection (which is a fairly complex object)

Comment (jwildfire) [inline src/init.js]: or perhaps we should just follow the webcharts paradigm and go with `wrap` - I also want to bind the raw data to the root object. Going to cut this code review short and make some commits.

**Comment by jwildfire** [comment]
PS - The modularization is *awesome* this is soooo much better than it was. 🎆 👏

**Comment by jwildfire** [comment]
@samussiah OK, made it maybe 1/3 of the way through the updates. I'm basically bringing this more in line with the patterns we use in webcharts - as such I did the following: 

- added a `chart` object beneath the root `aeTable` object. Now the only method for aeTable is `.createChart()` All other objects (controls, the various graphics elements) are nested under `chart`
So, to create a chart you will call `aeTable.createChart("body",data)` instead of just `aeTable("body",data)`
- attached the data and the wrap d3.selector to the chart object in `chart.init()` which allows ...
- ... greatly simplified function calls throughout. Basically, now we simply pass the `chart` itself everywhere. I'm maybe 30% through these updates at the moment. Honestly, this could almost certainly be simplified further, so that we would just use `this` throughout, but going to leave that for a later update.  
- Finally, I just added the event listeners to the controls `.init()` methods. This matches our new pattern much more closely (and is a lot easier to follow). 

Take a look and let me know if you have any feedback. Will be moving on to tackle the table functions in [here](https://github.com/RhoInc/aeexplorer/tree/refactor/src/chart/AETable) ASAP.

**Comment by jwildfire** [comment]
OK - I think I'm done with the refactor! Here's a [working example](http://bl.ocks.org/jwildfire/f74d38987750010b97e9e673c55de01a). I'll go through and write up some thorough change notes next week. In the meantime, can you guys (@brittsikora and @samussiah) poke at the example and see if you can break it. @samussiah would be good for you to review the code a bit too. 

Before we move towards a release candidate, I'd like to review the issue list with an eye towards potential breaking issues. We definitely want to avoid going to 4.0 anytime soon if we can.

**Comment by jwildfire** [comment]
Scratch that - custom filters are buggy in the current build 👎 I'll debug soon-ish.

**Comment by jwildfire** [comment]
Fixed the filters - it was an easy fix, thank goodness 😌. This is now officially ready for a QC by @brittsikora and code review from @samussiah.

**Comment by brittsikora** [comment]
I will do my best to break it :)

**Comment by jwildfire** [comment]
@qspencer Here's the PR for the current build of the Adverse Event Explorer for your review. 

Here is a [Working example](http://bl.ocks.org/jwildfire/raw/f74d38987750010b97e9e673c55de01a)
Here are the [related issues](https://github.com/RhoInc/aeexplorer/milestone/2)

Please make a branch off of `refactor` if you want to commit any code!

**Pull request merged**